select.js

```diff
@@ -1,14 +1,39 @@
-define(function() {
-    var init = function() {
-        $('body').on('change', '.c-select', function() {
-            var $container = $(this);
-            var value = $container.find('option:selected').text();
-
-            $container.find('.c-select__value').text(value);
-        });
-    };
+define(['$'], function($) {
+    //
+    // Component constructor
+    //
+    var Select = function Select($el, options) {
+        // Component instance properties. The `$` hash (by convention) caches
+        // the root node and any sub-components.
+        this.value = '';
+        this.dom = {
+            $root: $el,
+            $value: $el.find('.c-select__value')
+        };
+
+        // Event handlers should use $.proxy to preserve access to the instance
+        // as `this` within the handler.
+        this.dom.$root.on('change', $.proxy(this.update, this));
+
+        // Immediately invoke any methods needed to set up initial state.
+        this.update();
+    }
+
+    //
+    // Update the component’s visible value when the underlying select element’s
+    // value changes.
+    //
+    Select.prototype.update = function update() {
+        this.value = this.dom.$root.find('option:selected').text();
+        this.dom.$value.text(this.value);
+        this.dom.$root.trigger('select:update'); // notify subscribers
```

Comment:
@ry5n re: mimicking jQuery plugins it's also got to do with the fact that it's the most common plugin ecosystem our developers, partners and users deal with, so it reduces friction. How much does that matter? Not sure.

Using plugin hooks gives us some other advantages too:

- testing: mocking is easier with methods passed to the function rather than events
- static code analysis: one's IDE can follow code through, giving us design-time errors, etc.
- more discoverability: we can inspect the API provided by a plugin using plugin.prototype, but can't see a list of all the events it publishes/listens to, your IDE can list methods, etc.

However, events have the advantage of being able to set multiple handlers, and being more dynamic. I think an excellent middle ground is [signals](http://blog.millermedeiros.com/callbacks-promises-signals-and-events/). They maintain discoverability, testability, are typo-resistant, statically investigable (that's a word, right?), but also offer all the advantages of custom events.

Thoughts?